docs/sdk/v3/guides/liquidity/04-swap-and-add-liquidity.md

```diff
@@ -0,0 +1,97 @@
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L41
+```
+
+Please note that routing has been covered in greater detail in the [routing guide](../04-routing.md).
+### Constructing the parameters for the swap-and-add function call
+
+
+Having created the router, we now need to construct the parameters required to make a call to its `routeToRatio` function. This is exactly what we want in our example.
+
+The first parameter is an instance of `SwapAndAddConfig`, which sets configurations for the `routeToRatio ` algorithm. `ratioErrorTolerance` determines the margin of error the resulting ratio can have from the optimal ratio. `maxIterations` determines the maximum times the algorithm will iterate to find a ratio within error tolerance. If max iterations is exceeded, an error is returned:
+
+```js reference title="Constructing SwapAndAddConfig" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L43-L46
+```
+
+The second parameter, which is optional, is an instance of `SwapAndAddOptions`. If it is included, `routeToRatio` will return the calldata for executing the atomic swap-and-add. These options contain `swapConfig` and `addLiquidityOptions`. `swapConfig` configures to set a recipient of leftover dust from swap, `slippageTolerance` and a `deadline` for the swap.
+
+Then, `addLiquidityOptions` must contain a `tokenId` to add to an existing position, **or** a `recipient` to mint a new one. It also includes a `slippageTolerance` and `deadline` for adding liquidity.
+
+The only custom parameter for `swapAndAddOptions`, is the token id, which is no other than the position id of the last position we minted in the example:
+
+```js reference title="Constructing SwapAndAddOptions" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L48-L58
+```
+
+Thirdly, we construct two instances of `CurrencyAmount`s, each of which represents the initial balance of the token that we wish to swap-and-add, where the token is the token in our target liquidity pool. We use the configuration parameters of the guide to set those:
+
+```js reference title="Constructing the two CurrencyAmounts" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L60-L74
+```
+
+
+Finally we construct the position, a position object that contains the details of the position for which to add liquidity. The position liquidity can be set to 1, since liquidity is still unknown and will be set inside the call to `routeToRatio`:
```

Comment:
```suggestion
Finally we construct the position object that contains the details of the position for which to add liquidity. The position liquidity can be set to `1`, since liquidity is still unknown and will be set inside the call to `routeToRatio`:
```

Also, can we explain what setting liquidity to `1` does? Why does it need to be set if it's unknown? Does it act as a minimum or maximum, or will it be overridden?

Reply:
it's just a placeholder - adding that.